server/utils/nunjucksSetup.ts

```diff
@@ -116,10 +116,15 @@ export default function nunjucksSetup(app: express.Express, path: pathModule.Pla
       dev: true, // This is set to true to allow us to see the full stacktrace from errors in global functions, otherwise it gets swallowed and tricky to see in logs
     },
   )
-
+  njkEnv.addGlobal(
+    'callMacro',
+    function callMacro(macro: (context: Record<string, never>) => void, parameters: Record<string, never>) {
+      return macro({ ...this.ctx, ...parameters })
+    },
+  )
```

Comment:
Context can be added to macros by importing them using ` with context`. So, in a template that uses a `dateGds` macro, you would import it like so:

```jinja
{% from "partials/dateGds.njk" import dateGds with context %}
```

This would then make the context available inside the macro, where it can be accessed with:

```jinja
{% set context = fetchContext() %}
```

Possibly cleaner than introducing this extra macro wrapper? 

Another possibly cleaner approach would be to repurpose `callMacro` as a wrapper for the parameters themselves:


```suggestion
  njkEnv.addGlobal(
    'addContext',
    (parameters: Record<string, never>) => ({ ...this.ctx, ...parameters })
  )
```

which could then be called in the template like so:

```jinja
{{ dateGds(addContext(params)) }}
```